search/localsearch_nqueens.cpp

```diff
@@ -0,0 +1,184 @@
+			{
+				h++;
+				break;
+			}
+		}
+	}
+	//sum diagonal
+	for (size_t i = 0; i < NREINAS; i++)
+	{
+		aux_sol.push_back(solution_vector[i] - (i + 1));
+	}
+	for (size_t i = 0; i < NREINAS; i++)
+	{
+		for (size_t j = i + 1; j < NREINAS; j++)
+		{
+			if (aux_sol[i] == aux_sol[j])
+			{
+				h++;
+				break;
+			}
+		}
+	}
+	aux_sol.clear();
+	//sum inverse diagonal
+	for (size_t i = 0; i < NREINAS; i++)
+	{
+		aux_sol.push_back(solution_vector[i] + (i + 1));
+	}
+	for (size_t i = 0; i < NREINAS; i++)
+	{
+		for (size_t j = i + 1; j < NREINAS; j++)
+		{
+			if (aux_sol[i] == aux_sol[j])
+			{
+				h++;
+				break;
+			}
+		}
+	}
+	return h;
+}
+
+/**
+ * function: neighbour
+ * Generate a neighbor based on the current solution of
+ * the problem. Read about neighborhoods in local search.
+ * @param [in] solution_vector is a vector with the
+ * column number where the queens are located on the board.
+ * @returns a new vector similar to the one passed by parameter.
+ */
+std::vector<int> neighbour(std::vector<int> solution_vector) {
+	srand(time(nullptr)); //reset srand
+	int i = rand() % NREINAS + 1;
+	srand(time(nullptr)); //reset srand
+	int mov = (rand() % NREINAS + 1) - i;
+	solution_vector[i - 1] = (mov < 0) ? mov *= -1: (mov == 0) ? mov = 1 : mov;
+	return solution_vector;
+}
+
+/**
+ * function: print_solution
+ * Prints the solution obtained on the chessboard in console.
+ * @param [in] solution_vector is a vector with the
+ * column number where the queens are located on the board.
+ */
+void print_solution(std::vector<int> solution_vector) {
+	char chessboard[NREINAS][NREINAS] {};
+	for (int i = 0; i < NREINAS; i++)
+	{
+		chessboard[i][solution_vector[i] - 1] = 'Q';
+	}
+	for (int i = 0; i < NREINAS; i++)
+	{
+		for (int j = 0; j < NREINAS; j++)
+		{
+			if (chessboard[i][j] != 'Q')
+			{
+				std::cout << " _ ";
+			}
+			else { std::cout << " " << chessboard[i][j] << " ";}
+		}
+		std::cout << "\n";
+	}
+	std::cout << "\nSolucion: ";
+	for (int x : solution_vector)
+	{
+		std::cout << x;
+	}
+	std::cout << "\nh:" << cost(solution_vector) << std::endl;
+	std::cout << "\n";
+}
+
+/**
+ * function: searchrepeat
+ * Search for repeated neighbors (vector) to
+ * ensure the generation of new neighbors.
+ * @param [in] prob_neighbor is candidate to be a neighbor of the current solution.
+ * @param [in] neighborhood is vector of generated neighbors contained in memory.
+ */
+inline bool searchrepeat(std::vector<int> prob_neighbor, const std::vector<std::vector<int>>& neighborhood) {
+	return find(neighborhood.begin(), neighborhood.end(), prob_neighbor) != neighborhood.end();
+}
+
+/**
+ * Main function
+ * Heuristic Local Search - Rule first best neighbor
+ * - s = generate initial solution
+ * - while s not optimal local do{
+ * -	s* = N(s) with f(s) < f(s*)
+ * -	solution best of the neighbor
+ * -	s = s*
+ * -}
+ *
+ */
+int main() {
+	std::vector<int> init_solution = initial_solution();

```

Comment:
Since this is going to be the variable in which you modify and optimise the solution, I don't think it should be called `init_solution`
```suggestion
	std::vector<int> current_solution = initial_solution();

```

Reply:
Working¡